File: Cpp/Beautiful_Matrix.cpp
#include <iostream>
using namespace std;

int main() 
{
    int n,i;
    n=25;
    int a;
    for(i=0;i<n;i++)
    {
        cin >> a;
        if(a==1)
        {
            break;
        }
    }
    int row = i/5;
    int col = i%5;
    int res = abs(row-2)+abs(col-2);
    cout << res << endl;
    return 0;
}

File: Python/Beautiful_Matrix.py
br_flag = 0
for i in range(0,5):
    a = list(input().split(" "))
    for j in range(0, 5):
        if(int(a[j])):
            br_flag = 1
            break
    if(br_flag):
        break

res = abs(i-2)+abs(j-2)
print(res)